File: swift/[E]26.Remove-Duplicates-from-Sorted-Array.swift
/**
 * Runtime: 64 ms, faster than 79.22% of Swift online submissions for Remove Duplicates from Sorted Array.
 * Memory Usage: 14.9 MB, less than 48.16% of Swift online submissions for Remove Duplicates from Sorted Array.
 */
class Solution {
    func removeDuplicates(_ nums: inout [Int]) -> Int {
        if (nums.count == 0) {
            return 0
        }
        
        var lastNumber = nums[0]
        var index = 1
        for number in nums {
            if (lastNumber != number) {
                lastNumber = number
                nums[index] = number
                index += 1
            }
        }
        return index
    }
}

File: python/[E]26.Remove-Duplicates-from-Sorted-Array.py
# Runtime: 80 ms, faster than 82.51% of Python3 online submissions for Remove Duplicates from Sorted Array.
# Memory Usage: 15.8 MB, less than 82.79% of Python3 online submissions for Remove Duplicates from Sorted Array.
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums: 0
        else: 
            index = 1
            lastNum = nums[0]
            for number in nums:
                if number != lastNum:
                    lastNum = number
                    nums[index] = number
                    index += 1
            del nums[index:]
        
